ebi_eva_common_pyutils/metadata_utils.py

```diff
@@ -58,18 +61,67 @@ def get_dbsnp_mirror_db_info(pg_metadata_dbname, pg_metadata_user, pg_metadata_h
     return dbsnp_mirror_db_info
 
 
-def get_variant_warehouse_db_name_from_assembly_and_taxonomy(metadata_connection_handle, assembly, taxonomy):
-    query = f"select t.taxonomy_code, a.assembly_code " \
-            f"from assembly a " \
-            f"left join taxonomy t on (t.taxonomy_id = a.taxonomy_id) " \
-            f"where a.assembly_accession = '{assembly}'" \
-            f"and a.taxonomy_id = {taxonomy}"
+def get_taxonomy_code_from_taxonomy(metadata_connection_handle, taxonomy):
+    """
+    Retrieve an existing taxonomy code registered in the metadata database.
+    """
+    query = f"select t.taxonomy_code from  taxonomy t where t.taxonomy_id = {taxonomy}"
+    rows = get_all_results_for_query(metadata_connection_handle, query)
+    if len(rows) == 0:
+        return None
+    elif len(rows) > 1:
+        options = ', '.join(rows)
+        raise ValueError(f'More than one possible code for taxonomy {taxonomy} found: {options}')
+    return rows[0]
+
+
+def get_assembly_code_from_assembly(metadata_connection_handle, assembly):
+    """
+    Retrieve an existing assembly code registered in the metadata database.
+    """
+    query = f"select distinct assembly_code from assembly where assembly_accession={assembly};"
     rows = get_all_results_for_query(metadata_connection_handle, query)
     if len(rows) == 0:
         return None
     elif len(rows) > 1:
-        options = ', '.join((f'{r[0]}_{r[1]}' for r in rows))
-        raise ValueError(f'More than one possible database for assembly {assembly} and taxonomy {taxonomy} found: '
-                         f'{options}')
-    database_name = f'eva_{rows[0][0]}_{rows[0][1]}'
-    return database_name
+        options = ', '.join(rows)
+        raise ValueError(f'More than one possible code for assembly {assembly} found: {options}')
+    return rows[0]
+
+
+def build_variant_warehouse_database_name(taxonomy_code, assembly_code):
+    if taxonomy_code and assembly_code:
+        return f'eva_{taxonomy_code}_{assembly_code}'
+    return None
+
+
+def get_variant_warehouse_db_name_from_assembly_and_taxonomy(metadata_connection_handle, assembly, taxonomy):
+    """
+    Retrieve an existing database name by combining the taxonomy_code and assembly code registered in the metadata
+    database.
+    """
+    return build_variant_warehouse_database_name(
+        get_taxonomy_code_from_taxonomy(metadata_connection_handle, taxonomy),
+        get_assembly_code_from_assembly(metadata_connection_handle, assembly)
+    )
+
+
+def get_new_variant_warehouse_db_name_from_assembly_and_taxonomy(metadata_connection_handle, assembly, taxonomy):
```

Comment:
This subsumes the previous method. So I am wondering if we should still retain the other one.

Reply:
I was worried about backward compatibility when removing a function. There might also be cases where we only want to retrieve the database from EVAPRO and get `None` when it does not exists.

Reply:
Maybe should at least rename this method as the db name isn't necessarily new, right?  Though `get_new_or_existing_variant_warehouse_db_name_from_assembly_and_taxonomy` is a bit of a mouthful...

Reply:
I've simplified the function's name but kept them both for now.